drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -530,18 +530,21 @@ static const D3D12_RESOURCE_DIMENSION RD_TEXTURE_TYPE_TO_D3D12_RESOURCE_DIMENSIO
 	D3D12_RESOURCE_DIMENSION_TEXTURE2D,
 };
 
-void RenderingDeviceDriverD3D12::_resource_transition_batch(ResourceInfo *p_resource, uint32_t p_subresource, uint32_t p_num_planes, D3D12_RESOURCE_STATES p_new_state, ID3D12Resource *p_resource_override) {
+void RenderingDeviceDriverD3D12::_resource_transition_batch(ResourceInfo *p_resource, uint32_t p_subresource, uint32_t p_num_planes, D3D12_RESOURCE_STATES p_new_state) {
 	DEV_ASSERT(p_subresource != UINT32_MAX); // We don't support an "all-resources" command here.
 
 #ifdef DEBUG_COUNT_BARRIERS
 	uint64_t start = OS::get_singleton()->get_ticks_usec();
 #endif
 
 	ResourceInfo::States *res_states = p_resource->states_ptr;
-	ID3D12Resource *res_to_transition = p_resource_override ? p_resource_override : p_resource->resource;
 
 	if (p_new_state == D3D12_RESOURCE_STATE_COPY_DEST) {
 		res_states->subresource_states[p_subresource].init_state = ResourceInfo::States::ResourceStateEx::INIT_STATE_DONE;
+	} else if (p_new_state == D3D12_RESOURCE_STATE_UNORDERED_ACCESS) {
+		if (unlikely(!res_states->xfamily_fallback.subresources_dirty.is_empty())) {
+			res_states->xfamily_fallback.subresources_dirty[p_subresource / sizeof(uint64_t)] |= (uint64_t(1) << (p_subresource % sizeof(uint64_t)));
```

Comment:
It seems the intention of the code is to make a bitmask, but it's using the sizeof(uint64_t) which is in bytes. All of this seems to be effectively only using the first byte of each element of a vector of uint64_ts.

Reply:
Sounds good. I've sometimes even felt the temptation to use some bitmap class that some code in `thirdparty/` seems to provide... That said, I'll fix this by just using `64` in the expression and let's consider the bitset class an improvement for a future iteration.

Reply:
Good catch, by the way!